PopupContainer/PopupContainer.js

```diff
@@ -0,0 +1,206 @@
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		position: PropTypes.object,
+
+		/**
+		 * Whether rtl locale.
+		 *
+		 * @type {Boolean}
+		 * @private
+		 */
+		rtl: PropTypes.bool
+	},
+
+	defaultProps: {
+		position: {left: 0, top: 0}
+	},
+
+	styles: {
+		css,
+		className: 'container'
+	},
+
+	computed: {
+		style: ({position, rtl, forceDirection}) => {
+			if (forceDirection === 'rtl') {
+				return {transform: `translate(${position.left * (-1)}px,${position.top}px`};
+			}
+			if (forceDirection === 'ltr') {
+				return {transform: `translate(${position.left}px,${position.top}px`};
+			}
+			if (rtl) {
+				return {transform: `translate(${position.left * (-1)}px,${position.top}px`};
+			}
+			if (!rtl) {
+				return {transform: `translate(${position.left}px,${position.top}px`};
+			}
+		},
+		directionStyle: ({forceDirection, rtl}) => {
+			if (forceDirection) return {direction : forceDirection};
+			if (!forceDirection && rtl) return {direction: 'rtl'};
+			if (!forceDirection && !rtl) return {direction: 'ltr'};
+		}
+	},
+
+	render: ({children, className, directionStyle, style, ...rest}) => {
+		delete rest.forceDirection;
+		delete rest.position;
+		delete rest.rtl;
+
+		return (
+			<div style={directionStyle}>
+				<div className={css.PopupContainer}>
+					<div className={className} style={style} {...rest}>{children}</div>
+				</div>
+			</div>
+		);
+	}
+});
+
+// const I18nPopupContainer = I18nContextDecorator(
+//	 {rtlProp: 'rtl'},
+//	 PopupContainerBase
+// );
+
+/**
+ *
+ * @class PopupContainer
+ * @memberof sandstone/PopupContainer
+ * @extends sandstone/PopupContainerBase
+ * @ui
+ * @private
+ */
+class PopupContainer extends Component {
+
+	static propTypes = /** @lends sandstone/PopupContainer.PopupContainer.prototype */ {
+		/**
+		 * Force direction like rtl, ltr.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		forceDirection: PropTypes.string,
+
+		/**
+		 * Called when the user has attempted to close the popup.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onClose: PropTypes.func,
+
+		/**
+		 * Called when the popup is opened.
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		onOpen: PropTypes.func,
+
+		/**
+		 * Controls the visibility of the PopupContainer.
+		 *
+		 * By default, the PopupContainer and its contents are not rendered until `open`.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		open: PropTypes.bool,
+
+		/**
+		 * Position of the PopupContainer on the screen.
+		 *
+		 * @type {Object}
+		 * @public
+		 */
+		position: PropTypes.object
+	};
+
+	static defaultProps = {
+		open: false,
+		position: {left: 0, top: 0}
+	};
+
+	constructor (props) {
+		super(props);
+	}
+
+	handleClose = (ev) => {
+		forwardCustom('onClose')(ev, this.props);
+	};
+
+	handleOpen = (ev) => {
+		forward('onOpen', ev, this.props);
+	};
+
+	render () {
+		const {className, open, position, children, forceDirection, ...rest} = this.props;
```

Comment:
`position` and `forceDirection` can be passed to `PopupContainerBase` via `rest` if they are destructed here.

Reply:
Does this mean it should be deleted from this line? I want to explicitly show that `PopupContainerBase` should receive those props. what do you think?

Reply:
Some components in our libraries get a lot of props and some components just get one or two. So, we can't expect every component has intentionally exposed props in render block. That means, developers should not rely on exposed props. So, I believe we don't need to keep it as we cannot rely on it.

Reply:
I like it to be explicit for readability, but I agree with you. Currently, component unity seems more important. I cleaned up the properties.